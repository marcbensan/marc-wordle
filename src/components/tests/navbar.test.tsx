import Navbar from "@/components/navbar";
import {
  cleanup,
  fireEvent,
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import { beforeEach, expect, test, vi } from "vitest";

const mockPush = vi.fn();

vi.mock("next/navigation", () => ({
  useRouter: () => ({
    push: mockPush,
  }),
}));

beforeEach(() => {
  cleanup();
});

test("Match Navbar snapshot", () => {
  expect(render(<Navbar />)).toMatchSnapshot();
});

test("Renders the navbar with help icon", () => {
  render(<Navbar />);
  const helpIcon = screen.getByTestId("help-button");

  expect(helpIcon).toBeDefined();
});

test("Help dialog is initially closed", () => {
  render(<Navbar />);
  const dialogTitle = screen.queryByText(/How to Play/i);

  expect(dialogTitle).toBeNull();
});

test("Opens help dialog when help icon is clicked", async () => {
  render(<Navbar />);

  fireEvent.click(screen.getByTestId("help-button"));
  const dialogTitle = screen.getByText("How to Play");

  expect(dialogTitle).toBeDefined();
});

test("Displays all game rules in the dialog", async () => {
  render(<Navbar />);

  fireEvent.click(screen.getByTestId("help-button"));

  expect(screen.getByText(/Guess the WORDLE in 6 tries/i)).toBeDefined();
  expect(
    screen.getByText(/Each guess must be a valid 5-letter word/i)
  ).toBeDefined();
  expect(
    screen.getByText(
      /The color of the tiles will change to show how close your guess was to the word/i
    )
  ).toBeDefined();
});

test("Closes dialog when 'Got it' button is clicked", async () => {
  render(<Navbar />);

  fireEvent.click(screen.getByTestId("help-button"));

  expect(screen.getByText("How to Play")).toBeDefined();

  fireEvent.click(screen.getByText("Got it"));

  await waitFor(() => {
    expect(screen.queryByText("How to Play")).toBeNull();
  });
});

test("Redirects to home when the home button is clicked", () => {
  render(<Navbar />);
  fireEvent.click(screen.getByTestId("home-button"));

  expect(mockPush).toHaveBeenCalledWith("/");
});
