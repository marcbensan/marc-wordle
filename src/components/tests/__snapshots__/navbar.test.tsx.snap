// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Match Navbar snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full text-white"
      >
        <nav
          aria-label="Main"
          class="group/navigation-menu relative flex-1 items-center p-4 bg-wordle-dark w-full border-b border-wordle-secondary flex justify-end"
          data-orientation="horizontal"
          data-slot="navigation-menu"
          data-viewport="true"
          dir="ltr"
        >
          <div
            style="position: relative;"
          >
            <ul
              class="group flex-1 list-none justify-center gap-1 flex w-full items-center space-x-4"
              data-orientation="horizontal"
              data-slot="navigation-menu-list"
              dir="ltr"
            >
              <li
                class="relative flex flex-row space-x-1 items-center"
                data-slot="navigation-menu-item"
              >
                <button
                  class="cursor-pointer hover:opacity-80 transition-opacity"
                  data-testid="home-button"
                >
                  <svg
                    aria-hidden="true"
                    class="lucide lucide-house size-8"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8"
                    />
                    <path
                      d="M3 10a2 2 0 0 1 .709-1.528l7-5.999a2 2 0 0 1 2.582 0l7 5.999A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
                    />
                  </svg>
                </button>
              </li>
              <li
                class="relative flex items-center"
                data-slot="navigation-menu-item"
              >
                <button
                  aria-controls="radix-«r3»"
                  aria-expanded="false"
                  aria-haspopup="dialog"
                  class="cursor-pointer hover:opacity-80 transition-opacity"
                  data-slot="dialog-trigger"
                  data-state="closed"
                  data-testid="help-button"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="lucide lucide-circle-help size-8"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="10"
                    />
                    <path
                      d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
                    />
                    <path
                      d="M12 17h.01"
                    />
                  </svg>
                </button>
              </li>
            </ul>
          </div>
          <div
            class="absolute top-full left-0 isolate z-50 flex justify-center"
          />
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full text-white"
    >
      <nav
        aria-label="Main"
        class="group/navigation-menu relative flex-1 items-center p-4 bg-wordle-dark w-full border-b border-wordle-secondary flex justify-end"
        data-orientation="horizontal"
        data-slot="navigation-menu"
        data-viewport="true"
        dir="ltr"
      >
        <div
          style="position: relative;"
        >
          <ul
            class="group flex-1 list-none justify-center gap-1 flex w-full items-center space-x-4"
            data-orientation="horizontal"
            data-slot="navigation-menu-list"
            dir="ltr"
          >
            <li
              class="relative flex flex-row space-x-1 items-center"
              data-slot="navigation-menu-item"
            >
              <button
                class="cursor-pointer hover:opacity-80 transition-opacity"
                data-testid="home-button"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-house size-8"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8"
                  />
                  <path
                    d="M3 10a2 2 0 0 1 .709-1.528l7-5.999a2 2 0 0 1 2.582 0l7 5.999A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
                  />
                </svg>
              </button>
            </li>
            <li
              class="relative flex items-center"
              data-slot="navigation-menu-item"
            >
              <button
                aria-controls="radix-«r3»"
                aria-expanded="false"
                aria-haspopup="dialog"
                class="cursor-pointer hover:opacity-80 transition-opacity"
                data-slot="dialog-trigger"
                data-state="closed"
                data-testid="help-button"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-circle-help size-8"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <path
                    d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
                  />
                  <path
                    d="M12 17h.01"
                  />
                </svg>
              </button>
            </li>
          </ul>
        </div>
        <div
          class="absolute top-full left-0 isolate z-50 flex justify-center"
        />
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
